storage/uri.go

```diff
@@ -211,6 +211,42 @@ func Delete(u fyne.URI) error {
 
 }
 
+func DeleteRecursive(u fyne.URI) error {
+	list, err := List(u)
+
+	if err != nil {
+		return err
+	}
+
+	for _, v := range list {
+		listable, err := CanList(v)
+		if err != nil {
+			return err
+		}
+		if !listable {
+			err = Delete(v)
+			if err != nil {
+				return err
+			}
+			continue
+		} else {
+			err = DeleteRecursive(v)
+
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = Delete(u)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
No need to check the value and return the same value.
```suggestion
	return Delete(u)
```